elasticdl/python/worker/worker_pytorch.py

```diff
@@ -0,0 +1,684 @@
+        details about arguments of a worker.
+        """
+        self._worker_id = args.worker_id
+        self._job_type = args.job_type
+        self._minibatch_size = args.minibatch_size
+        self._log_loss_steps = args.log_loss_steps
+        (
+            model_inst,
+            self._dataset_fn,
+            self._loss,
+            self._opt_fn,
+            self._eval_metrics_fn,
+            self._prediction_outputs_processor,
+            self._custom_data_reader,
+            self._callbacks_list,
+        ) = get_model_spec(
+            model_zoo=args.model_zoo,
+            model_def=args.model_def,
+            dataset_fn=args.dataset_fn,
+            loss=args.loss,
+            optimizer=args.optimizer,
+            eval_metrics_fn=args.eval_metrics_fn,
+            model_params=args.model_params,
+            prediction_outputs_processor=args.prediction_outputs_processor,
+            custom_data_reader=args.custom_data_reader,
+            callbacks=args.callbacks,
+        )
+
+        self._collective_communicator = None
+        if (
+                self._distribution_strategy == DistributionStrategy.ALLREDUCE
+                and args.num_workers > 1
+        ):
+            self._collective_communicator = CollectiveCommunicator(
+                service_name=args.collective_communicator_service_name
+            )
+
+        self.set_model(model_inst)
+
+        self._model_version = -1
+        self._task_data_service = TaskDataService(
+            self._mc,
+            self._job_type == JobType.TRAINING_WITH_EVALUATION,
+            custom_data_reader=self._custom_data_reader,
+            data_reader_params=get_dict_from_params_str(
+                args.data_reader_params
+            ),
+            data_origin=args.training_data,
+        )
+        if self._dataset_fn is None:
+            if hasattr(
+                    self._task_data_service.data_reader, "default_dataset_fn"
+            ):
+                self._dataset_fn = (
+                    self._task_data_service.data_reader.default_dataset_fn()
+                )
+            else:
+                raise ValueError(
+                    "dataset_fn is required if the data_reader used does "
+                    "not provide default implementation of dataset_fn"
+                )
+        self._get_model_steps = args.get_model_steps
+        self._opt = self._opt_fn(model_inst)
+        self._model.optimizer = self._opt
+        self._non_embed_grads = {}
+        self._evaluation_result = {}
+
+    # TODO: Multiple tests are currently using this function to initialize
+    # self._model, where the initialization should be done via constructor.
+    def set_model(self, model_inst):
+        """Set model instance to worker."""
+        self._model = model_inst
+        self._model.float()
+        self._train_eagerly = False
+        # self._init_embeddings()
+
+    def _init_embedding_layer(self):
```

Comment:
Is ElasticDL Embedding supported by PyTorch implementation? If not yet, remove embedding related codes.

Reply:
ok